nova/bigvm/manager.py

```diff
@@ -267,15 +294,14 @@ def _get_providers(self, context):
             elif rp['uuid'] not in vmware_hvs:  # ignore baremetal
                 continue
             else:
-                # retrieve the MEMORY_MB resource
-                url = '/resource_providers/{}/inventories/{}'.format(
-                        rp['uuid'], MEMORY_MB)
-                resp = client.get(url)
-                if resp.status_code != 200:
-                    LOG.error('Could not retrieve inventory for RP %(rp)s.',
-                              {'rp': rp['uuid']})
+                # retrieve the MEMORY_MB and MEMORY_RESERVABLE_MB resources
+                memory_mb_inventory = self._get_provider_inventory(
+                    rp['uuid'], MEMORY_MB)
+                memory_reservable_mb_inventory = self._get_provider_inventory(
+                    rp['uuid'], MEMORY_RESERVABLE_MB_RESOURCE)
+                if (not memory_mb_inventory
+                    or not memory_reservable_mb_inventory):
```

Comment:
I was optimizing here, because we needed to get a single resource only. Instead of doing 2 requests, it might make sense to get the full inventory instead (in a single request) and just take the appropriate resource out of the dictionary.